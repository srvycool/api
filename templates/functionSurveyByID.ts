export const request = `
  {
    "version" : "2017-02-28",
    "operation" : "GetItem",
    "key" : {
      "node" : $util.dynamodb.toDynamoDBJson($ctx.stash.surveyID),
      "path" : $util.dynamodb.toDynamoDBJson("survey")
    }
  }
`;

export const response = `
  #set($typeMapping = {})
  $util.qr($typeMapping.put('MULTIPLE_CHOICE', 'SurveyMultipleChoice'))
  $util.qr($typeMapping.put('TEXTAREA', 'SurveyTextarea'))

  {
    "__typename": "$typeMapping[$ctx.result.surveyType]",
    "id": "$ctx.result.id",
    "label": "$ctx.result.label"
  }
`;
