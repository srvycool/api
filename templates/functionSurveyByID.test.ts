import Parser, { Context } from "appsync-template-tester";
import { request, response } from "./functionSurveyByID";

describe("templates:functionSurveyByID", () => {
  describe("request", () => {
    it("should match snapshot", () => {
      const parser = new Parser(request);
      const context = {
        stash: {
          surveyID: "aa-bb-cc",
        },
      } as Context;

      expect(parser.resolve(context)).toMatchSnapshot();
    });
  });

  describe("response", () => {
    describe("surveyType is multiple choice", () => {
      it("should match snapshot", () => {
        const parser = new Parser(response);
        const context = {
          result: {
            id: "aa-bb-cc",
            label: "Coffee?",
            surveyType: "MULTIPLE_CHOICE",
          },
        };

        expect(parser.resolve(context)).toMatchSnapshot();
      });
    });

    describe("surveyType is textarea", () => {
      it("should match snapshot", () => {
        const parser = new Parser(response);
        const context = {
          result: {
            id: "aa-bb-cc",
            label: "How are you?",
            surveyType: "TEXTAREA",
          },
        };

        expect(parser.resolve(context)).toMatchSnapshot();
      });
    });
  });
});
