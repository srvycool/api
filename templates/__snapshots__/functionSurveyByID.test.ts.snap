// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`function:surveyByID request should match snapshot 1`] = `
Object {
  "key": Object {
    "node": Object {
      "S": "aa-bb-ccdjfbnj",
    },
    "path": Object {
      "S": "survey",
    },
  },
  "operation": "GetItem",
  "version": "2017-02-28",
}
`;

exports[`function:surveyByID response surveyType is multiple choice should match snapshot 1`] = `
Object {
  "__typename": "SurveyMultipleChoice",
  "id": "aa-bb-cc",
  "label": "Coffee?",
}
`;

exports[`function:surveyByID response surveyType is textarea should match snapshot 1`] = `
Object {
  "__typename": "SurveyTextarea",
  "id": "aa-bb-cc",
  "label": "How are you?",
}
`;
